### PR TITLE
Finalize declaration based checks

### PR DESCRIPTION
**Pull request by jdstrand**
- finalize base declaration and adjust tests
- snapd-control should not auto-connect by default
- remove livepatch temporary code and tests
- remove hardcoded lxd-support checks
- remove hard-coded checks for docker
- remove hard-coded checks from docker-support
- add auto-connect test for livepatch-style plugs
- add plug installation tests

Submitting this now for wider review. Notes:
- there are no tests for the deny-connection rules yet
- the store has support for snap declarations, and as of 2016-10-14 they are added to all the known existing snaps

### CODE REVIEW

**Comment by jdstrand** [comment]
FYI, I added snap declarations to the store for:
```
docker:
plugs:
{
	"docker-support": {
		"allow-auto-connection": "true",
		"allow-installation": "true"
	}
}
slots:
{
	"docker": {
		"allow-connection": "true",
		"allow-installation": "true"
	}
}


lxd:
plugs:
{
	"lxd-support": {
		"allow-auto-connection": "true",
		"allow-installation": "true"
	}
}


snapweb:
plugs:
{
	"snapd-control": {
		"allow-auto-connection": "true",
		"allow-installation": "true"
	}
}


canonical-livepatch:
plugs:
{
	"hardware-observe": {
		"allow-auto-connection": "true"
	},
	"kernel-module-control": {
		"allow-auto-connection": "true",
		"allow-installation": "true"
	},
	"network-control": {
		"allow-auto-connection": "true"
	},
	"system-observe": {
		"allow-auto-connection": "true"
	}
}

uefi-fw-tools:
slots:
{
	"fwupd": {
		"allow-connection": "true"
	}
}


udisks2:
slots:
{
	"udisks2": {
		"allow-connection": "true"
	}
}


modem-manager:
slots:
{
	"modem-manager": {
		"allow-connection": "true"
	}
}


network-manager:
slots:
{
	"network-manager": {
		"allow-connection": "true"
	}
}

locationd:
slots:
{
	"location-control": {
		"allow-connection": "true"
	},
	"location-observe": {
		"allow-connection": "true"
	}
}

mir:
slots:
{
	"mir": {
		"allow-connection": "true"
	}
}


chrome-test:
plugs:
{
	"browser-support": {
		"allow-connection": {
	                    "plug-attributes": {
	                            "allow-sandbox": "true"
                         }
		}
	}
}


firefox-snap:
plugs:
{
	"browser-support": {
		"allow-connection": {
	                    "plug-attributes": {
	                            "allow-sandbox": "true"
                         }
		}
	}
}


vlc:
slots:
{
	"mpris": {
		"allow-connection": {
	                    "slot-attributes": {
	                            "name": "vlc"
                         }
		}
	}
}
```

That's everything I know of that are using slots or restricted plugs.

File: interfaces/builtin/basedeclaration.go [hunk 1]
```diff
@@ -129,28 +255,28 @@ slots:
...
   lxd-support:
-    allow-auto-connection:
-      plug-publisher-id:
-        - canonical
-      plug-snap-id:
-        - J60k4JY0HppjwOjW8dZdYc8obXKxujRu
     allow-installation:
       slot-snap-type:
         - core
+    deny-auto-connection: true
```
Comment (pedronis) [inline interfaces/builtin/basedeclaration.go]: should this one also day deny-connection: true ?

File: interfaces/builtin/basedeclaration.go [hunk 2]
```diff
@@ -50,25 +171,30 @@ slots:
...
   cups-control:
     allow-installation:
       slot-snap-type:
         - core
     deny-auto-connection: true
   docker:
+    allow-installation: false
+    deny-connection: true
     deny-auto-connection: true
   docker-support:
```
Comment (pedronis) [inline interfaces/builtin/basedeclaration.go]: should this one have deny-connection: true as well?

File: interfaces/builtin/basedeclaration.go [hunk 3]
```diff
@@ -112,7 +239,6 @@ slots:
     allow-installation:
       slot-snap-type:
         - core
-    deny-auto-connection: false
   kernel-module-control:
```
Comment (pedronis) [inline interfaces/builtin/basedeclaration.go]: should this one also have deny-connection: true?

File: interfaces/builtin/basedeclaration.go (hunk 2 above)
Comment (jdstrand) [inline interfaces/builtin/basedeclaration.go]: I was thinking that denying installation and auto connection was enough for the super-privileged implicit slots. Is there something I am missing?

File: interfaces/builtin/basedeclaration.go (hunk 3 above)
Comment (jdstrand) [inline interfaces/builtin/basedeclaration.go]: See docker-support comment, above.

**Comment by jdstrand** [comment]
FYI, per @pedronis request I updated bluez snap declaration in the store just now (note, it was during the test run).

**Comment by pedronis** [comment]
closing this, it was picked up and expanded with some more tests

**Pull request closed without merging**